internal/cloudrunci/cloudrunjobs.go

```diff
@@ -0,0 +1,324 @@
+		j.Image = fmt.Sprintf("gcr.io/%s/%s:%s", j.ProjectID, j.Name, runID)
+	}
+
+	if _, err := gcloud(fmt.Sprintf("%s: Building image %s", j.version(), j.Image), j.buildCmd()); err != nil {
+		return fmt.Errorf("gcloud: %s: %q", j.Image, err)
+	}
+	j.built = true
+
+	return nil
+}
+
+// Run starts the Job in Cloud Run Jobs.
+// This method will call Build and Create if necessary.
+func (j *RunJob) Run() error {
+	if err := j.validate(); err != nil {
+		return err
+	}
+	// Create() checks that the image was built
+	if !j.created {
+		if err := j.Create(); err != nil {
+			return err
+		}
+	}
+	if _, err := gcloud(fmt.Sprintf("%s: Running cloud run job", j.version()), j.runCmd()); err != nil {
+		return fmt.Errorf("gcloud: %v: %q", j.version(), err)
+	}
+	return nil
+}
+
+// Clean deletes the created Cloud Run service.
+func (j *RunJob) Clean() error {
+	// NOTE: don't check whether j.created is set.
+	// We may want to attempt to clean up if creation failed (i.e. to clean up the image).
+
+	if err := j.validate(); err != nil {
+		return err
+	}
+
+	if _, err := gcloud(fmt.Sprintf("%s: Deleting cloud run job", j.version()), j.deleteRunJobCmd()); err != nil {
+		return fmt.Errorf("gcloud: %v: %q", j.version(), err)
+	}
+	j.created = false
+
+	// If built is false, no image was created or is not managed by cloudrun-ci.
+	if j.built {
+		_, err := gcloud(fmt.Sprintf("%s: Deleting Image %s", j.version(), j.Image), j.deleteImageCmd())
+		if err != nil {
+			return fmt.Errorf("gcloud: %v: %q", j.version(), err)
+		}
+		j.built = false
+	}
+
+	return nil
+}
+
+func (j *RunJob) createCmd() *exec.Cmd {
+	args := append([]string{
+		"--quiet",
+		"alpha",
+		"run",
+		"jobs",
+		"create",
+		j.version(),
+		"--image",
+		j.Image,
+	}, j.CommonGCloudFlags()...)
+
+	if j.Env != nil {
+		for k := range j.Env {
+			args = append(args, "--set-env-vars", j.Env.Variable(k))
+		}
+	}
+
+	cmd := exec.Command(gcloudBin, args...)
+	cmd.Dir = j.Dir
+	return cmd
+}
+
+func (j *RunJob) buildCmd() *exec.Cmd {
+	args := []string{
+		"--quiet",
+		"builds",
+		"submit",
+		"--project",
+		j.ProjectID,
+	}
+	if j.AsBuildpack {
+		args = append(args, "--pack=image="+j.Image)
+	} else {
+		args = append(args, "--tag", j.Image)
+	}
+
+	cmd := exec.Command(gcloudBin, args...)
+	cmd.Dir = j.Dir
+	return cmd
+}
+
+// runCmd returns the gcloud command needed to start this RunJob
+func (j *RunJob) runCmd() *exec.Cmd {
+	args := append([]string{
+		"--quiet",
+		"alpha",
+		"run",
+		"jobs",
+		"run",
+		j.version(),
+		"--wait", // Waits for job to complete before returning.
+	}, j.CommonGCloudFlags()...)
+
+	cmd := exec.Command(gcloudBin, args...)
+	cmd.Dir = j.Dir
+	return cmd
+}
+
+func (j *RunJob) deleteImageCmd() *exec.Cmd {
+	args := []string{
+		"--quiet",
+		"container",
+		"images",
+		"delete",
+		j.Image,
+		"--force-delete-tags",
+	}
+
+	cmd := exec.Command(gcloudBin, args...)
+	cmd.Dir = j.Dir
+	return cmd
+}
+
+func (j *RunJob) deleteRunJobCmd() *exec.Cmd {
```

Comment:
I believe but need to check that you can't delete a job if an execution is currently running; therefore we may need to  delete the execution first

Reply:
Currently, Run() waits for the job to complete before returning. Should we (a) Not wait or (b) Make Wait optional, and probably default?